### ISSUE TITLE
More Options to Control Ordering for Bookmarks

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Super.Human.Portal#45]
For the bookmarks converted from the Prominic Portal, Justin would like to preserve the original ordering for both the categories and the bookmarks.  We will probably need to define a separate "Index" property for this.

The ordering should be:
- First by index
- Then alphabetical, if the indexes are the same

My current plan is to define the sorting on the agent side.  I am already doing this for the domino-49 example.   This will allow further customization, like allowing a configured view define the ordering.

I'm not sure if we want to show the index in the Royale UI  It would be better to allow sorting by drag/drop or arrow buttons, and then have the server compute an appropriate index.  This will probably be a separate update.

**Comment by JoelProminic** [comment]
I updated the CustomBookmarkRead agent to use the Bookmarks/FlatForAgent view.  This can be changed using the `bookmarks_view` configuration value.

**Comment by JoelProminic** [comment]
We made some updates to support this today.   I updated the [Bookmarks/FlatForAgent view](https://github.com/Moonshine-IDE/Super.Human.Portal/blob/main/Super.Human.Portal_DVE/nsfs/nsf-moonshine/odp/Views/Bookmarks_5cFlatForAgent.view) to sort by `index` first, then the category, and then the name.   If no index has been specified it will be treated as `0`, and displayed first in the UI.

The Royale interface was previously sorting the categories but not the bookmarks.   To support the above update, @piotrzarzycki21 removed the category sorting.   As a result, the category position will be based on the first appearance of the category in the bookmark list (I did a test with this to confirm this behavior).   I think this behavior is reasonable.  We could consider showing multiple categories with the same name in this case, but this could cause confusion when adding new bookmarks to that category.
